test/unit/Signer/Hmac/Blake2bTest.php

```diff
@@ -0,0 +1,90 @@
+<?php
+declare(strict_types=1);
+
+namespace Lcobucci\JWT\Signer\Hmac;
+
+use Lcobucci\JWT\Signer\Key\InMemory;
+use Lcobucci\JWT\SodiumBase64Polyfill;
+use PHPUnit\Framework\TestCase;
+
+use function sodium_crypto_generichash;
+use function sodium_crypto_generichash_keygen;
+
+/**
+ * @coversDefaultClass \Lcobucci\JWT\Signer\Hmac\Blake2b
+ *
+ * @uses \Lcobucci\JWT\SodiumBase64Polyfill::base642bin()
+ */
+final class Blake2bTest extends TestCase
+{
+    private string $key;
+
+    /** @before */
+    public function initializeKey(): void
+    {
+        $this->key = SodiumBase64Polyfill::base642bin(
+            'GOu4rLyVCBxmxP+sbniU68ojAja5PkRdvv7vNvBCqDQ=',
+            SodiumBase64Polyfill::SODIUM_BASE64_VARIANT_ORIGINAL
+        );
+    }
+
+    /**
+     * @test
+     *
+     * @covers ::algorithmId
+     */
+    public function algorithmIdMustBeCorrect(): void
+    {
+        $signer = new Blake2b();
+
+        self::assertEquals('BLAKE2B', $signer->algorithmId());
+    }
+
+    /**
+     * @test
+     *
+     * @covers ::sign
+     *
+     * @uses \Lcobucci\JWT\Signer\Key\InMemory
+     */
+    public function signMustReturnAHashAccordingWithTheAlgorithm(): string
+    {
+        $hash = sodium_crypto_generichash('test', $this->key);
+
+        self::assertSame($hash, (new Blake2b())->sign('test', InMemory::plainText($this->key)));
```

Comment:
I know this is a test but I think we should lead by example =p

```suggestion
        self::assertTrue(hash_equals($hash, (new Blake2b())->sign('test', InMemory::plainText($this->key))));
```

Reply:
That's copied from https://github.com/lcobucci/jwt/blob/62a0bf15f11493c140b3bf8681380696d23d172e/test/unit/Signer/HmacTest.php#L43

Maybe a better test should hardcode the hash, WDYT?

Reply:
Yeah, that's me from the past leading by the example (with wrong examples) 😅

Reply:
I, today, honestly wouldn't split singing and verification I'd different test methods.

On mobile now but will send a suggestion over in a few minutes.

Reply:
Thinking now I think pre-generated hash makes sense :smile: